app/jobs/channels/refresh_ms_oauth_token_job.rb

```diff
@@ -0,0 +1,62 @@
+# refer: https://gitlab.com/gitlab-org/ruby/gems/gitlab-mail_room/-/blob/master/lib/mail_room/microsoft_graph/connection.rb
+# refer: https://github.com/microsoftgraph/msgraph-sample-rubyrailsapp/tree/b4a6869fe4a438cde42b161196484a929f1bee46
+# https://learn.microsoft.com/en-us/azure/active-directory/develop/active-directory-configurable-token-lifetimes
+
+require 'microsoft_graph_auth'
+
+class Channels::RefreshMsOauthTokenJob < ApplicationJob
+  queue_as :low
+
+  def perform
+    Channel::Email.where(provider: 'microsoft').each do |channel|
+      # refresh the token here, with microsoft offline_access scope
+      provider_config = channel.provider_config || {}
+
+      refresh_tokens(channel, provider_config.with_indifferent_access) if provider_config[:refresh_token].present?
+    end
+  end
```

Comment:
let's move this to `services` instead. since we won't be using the perform block nor need to inherit the application job. We only need the access_token  method, right?

Reply:
Right! makes sense 👍🏼